aws/table_aws_ebs_snapshot.go

```diff
@@ -362,8 +332,9 @@ func buildEbsSnapshotFilter(ctx context.Context, d *plugin.QueryData, h *plugin.
 	if equalQuals["owner_id"] != nil {
 		ownerFilter.Name = aws.String("owner-id")
 		ownerFilter.Values = []string{equalQuals["owner_id"].GetStringValue()}
-	} else {
+	} else if equalQuals["owner_alias"] == nil && equalQuals["snapshot_id"] == nil {
```

Comment:
As discussed on a call earlier today, I think we should consolidate these `if` conditions to be `if equalQuals["owner_id"] == nil && equalQuals["owner_alias"] == nil && equalQuals["snapshot_id"] == nil`, as we essentially only want to set the `OwnerIds` input param to the calling account ID if none of these are specified.